lib/_http_server.js

```diff
@@ -825,7 +825,10 @@ const requestHeaderFieldsTooLargeResponse = Buffer.from(
 function socketOnError(e) {
   // Ignore further errors
   this.removeListener('error', socketOnError);
-  this.on('error', noop);
+
+  if (!this.hasListener('error', noop)) {
+    this.on('error', noop);
```

Comment:
Preventing new listeners from being added actually hides the user error as per https://github.com/nodejs/node/pull/43587#issuecomment-1402042926.

Reply:
I'm not sure I follow this. Are you referring to the user not closing the socket?

Reply:
Yes.

Reply:
I see. Well, the fix will hide the bad behavior but it will prevent the fix. It depends what we want to do...